### PR TITLE
fix: Gosec/sec fixes

### PR DESCRIPTION
**Pull request by ovsinc**
Here is a small set of security fixes. Security vulnerabilities discovered with gosec.
Fixed:
* Potential DoS vulnerability via decompression bomb
* Potential file inclusion via variable
* Implicit memory aliasing in for loop

### CODE REVIEW

File: object/role_upload.go [hunk 1]
```diff
@@ -43,6 +43,7 @@ func UploadRoles(owner string, fileId string) (bool, error) {
 
 	newRoles := []*Role{}
 	for index, line := range table {
+		line := line
```
Comment (hsluoyz) [inline object/role_upload.go:46]: What's this?

File: object/permission_upload.go [hunk 2]
```diff
@@ -43,6 +43,7 @@ func UploadPermissions(owner string, fileId string) (bool, error) {
 
 	newPermissions := []*Permission{}
 	for index, line := range table {
+		line := line
```
Comment (hsluoyz) [inline object/permission_upload.go:46]: What's this?

File: object/user_avatar_favicon.go [hunk 3]
```diff
@@ -124,6 +124,7 @@ func chooseFaviconLinkBySizes(links []Link) *Link {
 	var chosenLink *Link
 
 	for _, link := range links {
+		link := link
```
Comment (hsluoyz) [inline object/user_avatar_favicon.go:127]: What's this?

File: object/user_upload.go [hunk 4]
```diff
@@ -83,6 +83,7 @@ func UploadUsers(owner string, fileId string) (bool, error) {
 
 	newUsers := []*User{}
 	for index, line := range table {
+		line := line
```
Comment (hsluoyz) [inline object/user_upload.go:86]: What's this?

File: controllers/user_upload.go [hunk 5]
```diff
@@ -19,13 +19,14 @@ import (
...
 	"os"
+	"path/filepath"
 
 	"github.com/casdoor/casdoor/object"
 	"github.com/casdoor/casdoor/util"
 )
 
 func saveFile(path string, file *multipart.File) (err error) {
-	f, err := os.Create(path)
+	f, err := os.Create(filepath.Clean(path))
```
Comment (hsluoyz) [inline controllers/user_upload.go:29]: Does it work for at least Go 1.17?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: controllers/user_upload.go (hunk 5 above)
Comment (ovsinc) [reply controllers/user_upload.go:29]: Yes. It works on [go1.17.13](https://pkg.go.dev/path/filepath@go1.17.13#Clean)

File: object/user_upload.go (hunk 4 above)
Comment (ovsinc) [reply object/user_upload.go:86]: reassign the iteration variable inside the loop
see above

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**